### PR TITLE
Update function usage in SSR and remove need for commons-io

### PR DESCRIPTION
**Pull request by hafizrahman**
### Description

While looking at the Dependabot PR https://github.com/woocommerce/woocommerce-android/pull/13044 , I investigated a little bit and found that the dependency common-io was used just for a function in `WCSSRModelExt`.  Then I found from Stack Overflow a function that can replace it without the need for the dependency:

https://stackoverflow.com/a/5599842

This PR adds that function and unit tests, replaces the usage, and removes the dependency.

### Steps to reproduce

- Make sure unit tests pass
- Go to More Menu > Help & Support > System Status Report, and ensure these are showing correct values:
  - WordPress Environment > WP Memory Limit
  - Server Environment > PHP Post Max Size
  - Server Environment > Max Upload Size

### Testing information

Nothing specific needed here.

### The tests that have been performed

I checked the steps listed above.

### Images/gif

n/a

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/WCSSRModelExt.kt [hunk 1]
```diff
@@ -134,7 +133,7 @@ private fun formatEnvironmentData(data: JSONObject): String {
 
     val maxUploadSize = data.optString("max_upload_size", MISSING_VALUE)
     if (maxUploadSize != MISSING_VALUE) {
-        sb.append("PHP Post Max Size: ${byteCountToDisplaySize(maxUploadSize.toLong())}\n")
+        sb.append("Max Upload Size: ${maxUploadSize.readableFileSize()}\n")
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/WCSSRModelExt.kt:136]: Did you also fix a bug here with the change `PHP Post Max Size` → `Max Upload Size`?

**Review by irfano — COMMENTED** [review]
I'm thinking if we should use `com.zendesk.util.FileUtils.humanReadableFileSize.humanReadableFileSize()` since we're already using this Zendes library for other purposes. This would save us from adding new code to the repository, but it would still use an unnecessary dependency. I wonder about your opinion.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/WCSSRModelExt.kt (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/WCSSRModelExt.kt:136]: Yes, this was an incorrect label that went unnoticed before (should be "Max upload size").

**Comment by hafizrahman** [comment]
> I'm thinking if we should use com.zendesk.util.FileUtils.humanReadableFileSize.humanReadableFileSize() since we're already using this Zendes library for other purposes. This would save us from adding new code to the repository, but it would still use an unnecessary dependency. I wonder about your opinion.

Nice find, I did not know about that before. For this purpose I think it's better to not add dependency as it would save time in the future dealing with dependabot. I don't think the added new code will need a lot of maintenance, and it's also only used in one small part of the app. What do you think?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt [hunk 2]
```diff
@@ -100,3 +105,28 @@ fun String.toCamelCase(delimiter: String = " "): String {
...
+ *
+ * Source: https://stackoverflow.com/a/5599842
+ *
+ * Examples:
+ * "1024".readableFileSize() -> "1 kB"
+ * "1500000".readableFileSize() -> "1.4 MB"
+ * "0".readableFileSize() -> "0"
+ * Invalid input returns "0"
+ *
+ *
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt]: np: Can we remove the double line breaks? A single break matches the code style of comments.

**Review by irfano — APPROVED** [review]
> For this purpose I think it's better to not add dependency as it would save time in the future dealing with dependabot. I don't think the added new code will need a lot of maintenance, and it's also only used in one small part of the app. What do you think?

I agree. Let’s proceed with adding this PR. I’ve approved it, but I won’t merge it yet to give you a chance to address the feedback I provided.

I noticed a bug in the current release. We're rounding 1.99 GB to 1 GB. We're also fixing it with this PR. I guess it's not a critical issue and no need to report it to HEs.

|before|after|web|
|-|-|-|
|||Screenshot 2024-12-10 at 18 13 11|

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt]: Good catch, handled in https://github.com/woocommerce/woocommerce-android/pull/13073/commits/c616faf6ab1b020567aff9e77393b6e6e2f386f9

**Comment by hafizrahman** [comment]
> I noticed a bug in the current release. We're rounding 1.99 GB to 1 GB. We're also fixing it with this PR. I guess it's not a critical issue and no need to report it to HEs.

Great catch, thanks for testing so thoroughly! I think it's nothing we should announce, but great that it's fixed as a side effect.

**Pull request merged**